plugins/jira-dashboard-backend/src/service/router.ts

```diff
@@ -215,6 +217,51 @@ export async function createRouter(
     },
   );
 
+  router.get('/dashboards/userIssues', async (request, response) => {
+    const { token } = await auth.getPluginRequestToken({
+      onBehalfOf: await auth.getOwnServiceCredentials(),
+      targetPluginId: 'catalog',
+    });
+
+    const credentials = await httpAuth.credentials(request);
+    const info = await userInfo.getUserInfo(credentials);
+
+    const userEntity = (await catalogClient.getEntityByRef(info.userEntityRef, {
+      token,
+    })) as UserEntity;
+
+    if (!userEntity?.spec?.profile?.email) {
+      const error = `User email cannot be extracted`;
+      logger.info(error);
+      response.status(400).json(error);
+      return;
+    }
+
+    const email = userEntity.spec.profile.email;
+    if (!email || email.trim() === '') {
+      const error = `User email is not defined`;
+      logger.info(error);
+      response.status(400).json(error);
+      return;
+    }
+
+    // maybe make it configurable to use email as username ?
+    const username = email.substring(0, email.indexOf('@'));
```

Comment:
There is already functionality for getting the user email, in[ filters.ts](https://github.com/AxisCommunications/backstage-plugins/blob/main/plugins/jira-dashboard-backend/src/filters.ts#L24). 

Rather than handling this differently here, should we try to extract this functionality into it's own function and use it here?

Reply:
@fridajac  note this code is extracting prefix - before '@',  not suffix , but maybe the fallback to the entity.name might be a good idea

Reply:
Ah, I get it now how it works. Yes, some function might be handy. Shall I make it? Into the filters.ts ?

Reply:
To spare your time with answering, I made that function :-)...

Reply:
Nice! Looks good :+1: Great that you wrote a description of the method in the comment, much better than before!